packages/x-components/src/x-modules/tagging/components/__tests__/tagging.spec.ts

```diff
@@ -0,0 +1,116 @@
+import { mount, Wrapper } from '@vue/test-utils';
+import Vue from 'vue';
+import { Component, Prop, Provide } from 'vue-property-decorator';
+import { installNewXPlugin } from '../../../../__tests__/utils';
+import { getXComponentXModuleName, isXComponent } from '../../../../components/x-component.utils';
+import { XComponentBusAPI } from '../../../../plugins/x-plugin.types';
+import { baseSnippetConfig } from '../../../../views/base-config';
+import { SnippetConfig } from '../../../../x-installer/api/api.types';
+import Tagging from '../../../tagging/components/tagging.vue';
+
+@Component({
+  template: `<div><slot/></div>`
+})
+class Provider extends Vue {
+  @Prop({ default: baseSnippetConfig })
+  @Provide()
+  public snippetConfig!: SnippetConfig;
+}
+
+function renderTagging({
+  template = `<Provider :snippetConfig="snippetConfig"><Tagging :consent="consent"/></Provider>`,
+  consent = undefined,
+  snippetConfig = baseSnippetConfig
+}: RenderTaggingOptions = {}): RenderTaggingAPI {
+  const [, localVue] = installNewXPlugin();
+
+  const wrapper = mount(
+    {
+      components: {
+        Tagging,
+        Provider
+      },
+      props: ['consent', 'snippetConfig'],
+      template
+    },
+    {
+      propsData: {
+        consent,
+        snippetConfig
+      },
+      localVue
+    }
+  );
+
+  return {
+    wrapper: wrapper.findComponent(Tagging),
+    on: wrapper.vm.$x.on.bind(wrapper.vm.$x)
```

Comment:
I'll remove this `on` and use the wrapper in the rest of the tests:

```javascript
const { wrapper } = renderTagging({ consent: true });
const eventSpy = jest.fn();

await wrapper.vm.$x.on('ConsentProvided').subscribe(eventSpy);
expect(eventSpy).toHaveBeenNthCalledWith(1, true);
```

Reply:
ill prefer to return the on in the function, like in the url tests